### PR TITLE
Issue #6586: aligned javadoc/xdoc for NoClone

### PR DESCRIPTION
**Pull request by pbludov**
Issue #6586 (part of #5750)

No changes in the code.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoCloneCheck.java [hunk 1]
```diff
@@ -29,12 +29,14 @@
  * Checks that the clone method is not overridden from the
  * Object class.
  * </p>
- *
- * <p>Rationale: The clone method relies on strange/hard to follow rules that
- * do not work it all situations.  Consequently, it is difficult to
- * override correctly.  Below are some of the rules/reasons why the clone
- * method should be avoided.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoCloneCheck.java]: Why remove the last sentence and just place it before we start the list?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoCloneCheck.java [hunk 2]
```diff
@@ -79,42 +81,57 @@
...
  * Just keep in mind if this type of polymorphic cloning is required
  * then a properly implemented clone method may be the best choice.
- *
+ * </p>
  * <p>Much of this information was taken from Effective Java:
  * Programming Language Guide First Edition by Joshua Bloch
  * pages 45-52.  Give Bloch credit for writing an excellent book.
  * </p>
+ * <p>
+ * This check is almost exactly the same as the {@code NoFinalizerCheck}.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoCloneCheck.java]: Should this and the `See` go before the raionale?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoCloneCheck.java (hunk 1 above)
Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoCloneCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoCloneCheck.java (hunk 2 above)
Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoCloneCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoCloneCheck.java (hunk 1 above)
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoCloneCheck.java]: @pbludov Please point to me where `Below are some of the rules/reasons why the clone method should be avoided.` was re-added?

**Comment by rnveach** [comment]
@pbludov Please remember to reply when you make a change. I didn't get a notification so I didn't realize this was done.
